libs/category/driver/magento/src/category.service.spec.ts

```diff
@@ -250,6 +256,123 @@ describe('Driver | Magento | Category | CategoryService', () => {
     });
   });
 
+  describe('getByUri | getting a category by URI', () => {
+    let mockCategoryUriRequest: DaffCategoryUriRequest;
+    let uri: string;
+    let result: Observable<DaffGetCategoryResponse>;
+
+    beforeEach(() => {
+      uri = 'uri';
+      mockCategoryUriRequest = {
+        kind: DaffCategoryRequestKind.URI,
+        uri: `${uri}.html`,
+      };
+      result = categoryService.getByUri(mockCategoryUriRequest);
+    });
+
+    it('should return a category with the correct info', fakeAsync(() => {
+      result.subscribe(res => {
+        expect(res.category.id).toEqual(String(mockMagentoCategory.id));
+        expect(res.category.name).toEqual(String(mockMagentoCategory.name));
+        flush();
+      });
+
+      const categoryOp = controller.expectOne(MagentoGetCategoryQuery);
+      const filterTypesOp = controller.expectOne(MagentoGetCategoryFilterTypes);
+
+      categoryOp.flushData(mockGetCategoryResponse);
+      filterTypesOp.flushData(mockGetFilterTypesResponse);
+
+      tick();
+
+      const productsOp = controller.expectOne(MagentoGetProductsQuery);
+      productsOp.flushData(mockGetProductsResponse);
+    }));
+
+    it('should query the category with the truncated URI', fakeAsync(() => {
+      result.subscribe();
+
+      const categoryOp = controller.expectOne(MagentoGetCategoryQuery);
+      const filterTypesOp = controller.expectOne(MagentoGetCategoryFilterTypes);
+
+      categoryOp.flushData(mockGetCategoryResponse);
+      filterTypesOp.flushData(mockGetFilterTypesResponse);
+
+      tick();
+
+      const productsOp = controller.expectOne(MagentoGetProductsQuery);
+      productsOp.flushData(mockGetProductsResponse);
+
+      expect(categoryOp.operation.variables.filters.url_path.eq).toEqual(uri);
+
+      flush();
+    }));
+
+    it('should query the products with the category ID', fakeAsync(() => {
+      result.subscribe();
+
+      const categoryOp = controller.expectOne(MagentoGetCategoryQuery);
+      const filterTypesOp = controller.expectOne(MagentoGetCategoryFilterTypes);
+
+      categoryOp.flushData(mockGetCategoryResponse);
+      filterTypesOp.flushData(mockGetFilterTypesResponse);
+
+      tick();
+
+      const productsOp = controller.expectOne(MagentoGetProductsQuery);
+      productsOp.flushData(mockGetProductsResponse);
+
+      expect(productsOp.operation.variables.filter.category_id.eq).toEqual(String(mockMagentoCategory.id));
+
+      flush();
+    }));
+
+    describe('when filters are requested', () => {
```

Comment:
I think this should read "when the request applied a filter" or something like that. Filters are always requested/retrieved from the backend.

Reply:
changed